### DISCUSSION TITLE
Can you share the design of RBAC？

### DISCUSSION THREAD

**Discussion by meidea** [logto-io/logto#2451] [General]

**Comment by gao-sun** [comment]
Thanks for submitting your question. Please note the discussion page also follows our issue guideline which requires an English title and body.

It'll be great if you can update the content before we have further conversations. Cheers.

**Comment by fleuraly** [comment]
Hey @meidea , we are building the first version of RBAC. For the MVP version, we support `role` and `permission (Scope)`.  For the overall structure, the short answer is: The `permission(scope) `is under API resource and you can create it on your own. `Role` is a first-level entity. You can create roles and then allocate permissions (across APIs possible) to each role, which also can be assigned to specific users.

Here are the features we are proposing,
1. Assign a default role to general users
2. Create and delete permissions in API resource
3. Create roles in a section called "Access control"
4. Assign and remove roles from users
5. Assign and remove permissions from roles

 Let us know if you have any other questions!

**Reply by meidea** [reply]
@fleuraly Does permission support tree structure ?

API Resource:   https://api.logto.io

Permissions:
```
Applications:
    List
    Create Application
    Modify Application
```
